src/Uno.UI/UI/Xaml/UIElement.Layout.netstd.cs

```diff
@@ -131,13 +237,93 @@ private void MeasureVisualTreeRoot(Size availableSize)
 
 		private void DoMeasure(Size availableSize)
 		{
-			InvalidateArrange();
+			var isDirty =
+				(availableSize != LastAvailableSize)
+				|| IsMeasureDirty
+				|| !GetLayoutFlag(LayoutFlag.FirstMeasureDone);
+
+			if (!isDirty && !IsMeasureDirtyPath)
+			{
+				return; // Nothing to do
+			}
+
+			if (!GetLayoutFlag(LayoutFlag.FirstMeasureDone))
+			{
+				SetLayoutFlag(LayoutFlag.FirstMeasureDone);
+				isDirty = true;
+			}
 
-			// We must reset the flag **BEFORE** doing the actual measure, so the elements are able to re-invalidate themselves
-			_isMeasureValid = true;
+			var remainingTries = MaxLayoutIterations;
 
-			MeasureCore(availableSize);
-			LayoutInformation.SetAvailableSize(this, availableSize);
+			while (--remainingTries > 0)
+			{
+				if (isDirty)
+				{
+					// We must reset the flag **BEFORE** doing the actual measure, so the elements are able to re-invalidate themselves
+
+					ClearLayoutFlags(LayoutFlag.MeasureDirty | LayoutFlag.MeasureDirtyPath);
+
+					// The dirty flag is explicitly set on this element
+#if DEBUG
+					try
+#endif
+					{
+						MeasureCore(availableSize);
+						InvalidateArrange();
+					}
+#if DEBUG
+					catch (Exception ex)
+					{
+						_log.Error($"Error measuring {this}", ex);
+					}
```

Comment:
I think that we should re-throw in order to have somehow the same behavior as in RELEASE
```suggestion
						throw;
					}
```